janusgraph-codepipelines-ci/README.md

```diff
@@ -0,0 +1,57 @@
+#JanusGraph CodePipelines CI
+CodePipelines CI is a mechanism JanusGraph can use to do release testing in massively
+parallel fashion (to the extent of AWS CodePipelines and CodeBuild service limits).
+
+## Steps to kick off a build
+Follow the procedure below to run all release testing in parallel.
+This procedure requires you to have an AWS account.
+It also requires you to create two service roles in IAM: one for CodePipeline and
+one for CodeBuild.
+
+1. Get a personal access token from [GitHub](https://github.com/settings/tokens) with repo and admin:repo_hook scopes.
+2. Navigate to the [AWS Console](https://console.aws.amazon.com) and create an IAM User with the following managed policies: AmazonS3FullAccess, AWSCodePipelineFullAccess, AWSCodeBuildAdminAccess.
+3. For this user, create security credentials and then register them in the `code-pipelines` profile on your computer with `aws configure --profile code-pipelines`.
+4. Create a service role for CodeBuild as described [here](http://docs.aws.amazon.com/codebuild/latest/userguide/setting-up.html#setting-up-service-role).
+5. Create an IAM policy for a CodePipelines service role as described [here](http://docs.aws.amazon.com/codepipeline/latest/userguide/iam-identity-based-access-control.html#view-default-service-role-policy).
+6. Create a service role for CodePipelines by using the policy you created in step 5 and then set the STS trust to codepipeline.amazonaws.com, like what you did for CodeBuild.
+7. Build and run the program to create the stack and start a build. Replace:
+```bash
+mvn clean package
+#define variables
+export AWS_PROFILE_NAME='profile_to_use'
+export AWS_ACCOUNT_NUMBER='000000000000'
+export AWS_CODEBUILD_ROLE='codebuild_role_name'
+export AWS_CODEPIPELINE_ROLE='codepipeline_role_name'
+export GITHUB_TOKEN='aaaa'
+
+#kick off regular tests
+java -jar target/janusgraph-codepipelines-ci-0.2.0-SNAPSHOT.jar \
+--region us-west-2 \
+--bucket jgamcp \
+--github-owner amcp \
+--github-repo janusgraph \
+--github-branch codepipelines \
+--profile ${AWS_PROFILE_NAME} \
+--codebuild-role-arn arn:aws:iam::${AWS_ACCOUNT_NUMBER}:role/${AWS_CODEBUILD_ROLE} \
+--codepipeline-role-arn arn:aws:iam::${AWS_ACCOUNT_NUMBER}:role/${AWS_CODEPIPELINE_ROLE} \
+--github-token ${GITHUB_TOKEN} \
+--pipelines ${PWD}/pipe.json
+
+#kick off tinkerpop tests
```

Comment:
Consider removing the second call here. It's a complicated call and the only difference is the pipeline used. Instead create another section or discussion (or link) on how to create the pipelines files and indicate that examples are included to run default (pipe.json) and TinkerPop (tp-pipe.json) tests.